src/koga/archive.lisp

```diff
@@ -40,4 +40,70 @@
                                        tar-name temp-path))))
      :want-stream-p nil)
     (message nil "Compressing archive...")
-    (run-program (format nil "gzip ~a" tar-name))))
+    (run-program (format nil "gzip ~a" tar-name))
+    (generate-ebuild version gz-name)))
+
+(defun generate-ebuild (&optional (version "9999") archive)
+  (labels ((mk-os-path (&rest args)
+             ;; args: (full-os-path) | (dir fname) | (dir base ext)
+             (format nil "~<~^~a~^/~a~^.~a~:>" args))
+           (os-path-fname (path)
+             (nth-value 2 (uiop:split-unix-namestring-directory-components path)))
+           (os->cl-path (os-path)
+             (uiop:ensure-absolute-pathname os-path (uiop:getcwd)))
+           (fbytes (os-path)
+             (or (with-open-file (s (os->cl-path os-path) :element-type 'unsigned-byte
+                                                          :if-does-not-exist nil)
+                   (when s (file-length s)))
+                 0))
+           (calc-hash (os-path hash-type)
+             (subseq (uiop:run-program (format nil "cksum -a ~a --untagged \"~a\"" os-path hash-type)
+                                       :output :string)
+                     0 128))
+           (copy-f (src-os-path dest-os-path)
+             (let ((s-path (os->cl-path src-os-path))
+                   (d-path (os->cl-path dest-os-path)))
+               (uiop:ensure-all-directories-exist (list d-path))
+               (uiop:delete-file-if-exists d-path)
+               (uiop:copy-file s-path d-path))))
+    (let* ((src-dir "gentoo")
+           (dest-dir "gentoo/clasp")
+           (manifest-name "Manifest")
+           (pkg-ver (concatenate 'string "clasp-" version))
+           ;; Files to copy: each entry (manifest-label source dest)
+           (flist `(,@(unless (equal version "9999")
+                        `(("EBUILD" ,(mk-os-path src-dir "ebuild.template")
+                                    ,(mk-os-path dest-dir pkg-ver "ebuild"))))
+                    ("EBUILD" ,(mk-os-path src-dir "ebuild.template")
+                              ,(mk-os-path dest-dir "clasp-9999.ebuild"))
+                    ("MISC" ,(mk-os-path src-dir "metadata.xml")
+                            ,(mk-os-path dest-dir "metadata.xml"))
+                    ,@(when archive `(("DIST" nil ,(mk-os-path archive))))))
+           ;; --
+           (manifest-path (os->cl-path (mk-os-path dest-dir manifest-name)))
+           (manifest-temp-path (uiop:tmpize-pathname manifest-path)))
+      (message nil "Creating Gentoo/Portage distro files for ~a..." pkg-ver)
+      (uiop:ensure-all-directories-exist (list manifest-temp-path))
+      (uiop:with-output-file (mfst-dest manifest-temp-path :if-exists :supersede)
```

Comment:
Please don't use the uiop:with-output-file/uiop:with-input-file functions. They are just syntactic sugar.

Reply:
Done